### PR TITLE
Initial release

### PR DESCRIPTION
**Pull request by frasdav**

### CODE REVIEW

**Review by CDA0 — DISMISSED** [review]

File: pkg/cmd/app/apply/azurerm.go [hunk 1]
```diff
@@ -0,0 +1,1594 @@
...
+	"golang.org/x/exp/slices"
+)
+
+//go:embed default_role_management_policy.json
+var defaultRoleManagementPolicyPropertiesData string
+
+// var defaultRoleManagementPolicyProperties armauthorization.RoleManagementPolicyProperties
+
+func init() {
+	// roleManagementPolicyProperties := armauthorization.RoleManagementPolicyProperties{}
```
[the remark this answers is not in the document]
Comment (frasdav) [reply pkg/cmd/app/apply/azurerm.go:38]: Fixed.

File: pkg/cmd/cli/apply/azurerm.go [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+	cmd.Flags().StringVarP(&configDir, "config-dir", "c", wd, "Config directory")
+	cmd.Flags().BoolVarP(&planOnly, "plan-only", "p", false, "Plan-only")
+	cmd.Flags().StringVarP(&subscriptionId, "subscription-id", "s", "", "Subscription Id") // TODO: Support name
+
+	cobra.MarkFlagRequired(cmd.Flags(), "subscription-id")
+
+	return cmd
+}
+
+func PrintHeader(configDir string, scope string) {
```
[the remark this answers is not in the document]
Comment (frasdav) [reply pkg/cmd/cli/apply/azurerm.go:49]: Fixed.

File: pkg/cmd/cli/apply/azurerm.go [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+	subscriptionId string
+)
+
+// NewCmdApplyAzureRm creates a command to apply the Azure RM config
+func NewCmdApplyAzureRm() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "azurerm",
+		Short: "Apply Azure Resource Manager config",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			PrintHeader(configDir, subscriptionId)
```
[the remark this answers is not in the document]
Comment (frasdav) [reply pkg/cmd/cli/apply/azurerm.go:25]: Fixed.

File: pkg/cmd/cli/plan/azurerm.go [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+	subscriptionId string
+)
+
+// NewCmdPlanAzureRm creates a command to llan the Azure RM config changes
+func NewCmdPlanAzureRm() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "azurerm",
+		Short: "Plan Azure Resource Manager config changes",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			PrintHeader(configDir, subscriptionId)
```
[the remark this answers is not in the document]
Comment (frasdav) [reply pkg/cmd/cli/plan/azurerm.go:24]: Fixed.

File: pkg/cmd/cli/plan/azurerm.go [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+
+	cmd.Flags().StringVarP(&configDir, "config-dir", "c", wd, "Config directory")
+	cmd.Flags().StringVarP(&subscriptionId, "subscription-id", "s", "", "Subscription Id") // TODO: Support name
+
+	cobra.MarkFlagRequired(cmd.Flags(), "subscription-id")
+
+	return cmd
+}
+
+func PrintHeader(configDir string, scope string) {
```
[the remark this answers is not in the document]
Comment (frasdav) [reply pkg/cmd/cli/plan/azurerm.go:47]: Fixed.

File: pkg/cmd/cli/validate/azurerm.go [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+	configDir string
+)
+
+// NewCmdValidate creates a command to validate the Azure Rm config
+func NewCmdValidateAzureRm() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "azurerm",
+		Short: "Validate Azure RM config",
+		RunE: func(_ *cobra.Command, _ []string) error {
+			PrintHeader(configDir)
```
[the remark this answers is not in the document]
Comment (frasdav) [reply pkg/cmd/cli/validate/azurerm.go:23]: Fixed.

File: pkg/cmd/cli/validate/azurerm.go [hunk 7]
```diff
@@ -0,0 +1,51 @@
...
+	if err != nil {
+		panic(err)
+	}
+
+	cmd.Flags().StringVarP(&configDir, "config-dir", "c", wd, "Config directory")
+
+	return cmd
+}
+
+func PrintHeader(configDir string) {
```
[the remark this answers is not in the document]
Comment (frasdav) [reply pkg/cmd/cli/validate/azurerm.go:43]: Fixed.

**Review by CDA0 — APPROVED** [review]

**Review by CDA0 — APPROVED** [review]

**Pull request merged**